geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/StatSerializer.scala

```diff
@@ -71,53 +71,94 @@ object KryoStatSerializer {
-      case HistogramByte     => readHistogram(input, sft, immutable)
-      case FrequencyByte     => readFrequency(input, sft, immutable, 2)
-      case Z3HistogramByte   => readZ3Histogram(input, sft, immutable, 2)
-      case Z3FrequencyByte   => readZ3Frequency(input, sft, immutable, 2)
-      case IteratorStackByte => readIteratorStackCount(input, immutable)
-      case SeqStatByte       => readSeqStat(input, sft, immutable)
-      case FrequencyByteV1   => readFrequency(input, sft, immutable, 1)
-      case Z3HistogramByteV1 => readZ3Histogram(input, sft, immutable, 1)
-      case Z3FrequencyByteV1 => readZ3Frequency(input, sft, immutable, 1)
+      case CountByte           => readCount(input, immutable)
+      case MinMaxByte          => readMinMax(input, sft, immutable)
+      case EnumerationByte     => readEnumeration(input, sft, immutable)
+      case TopKByte            => readTopK(input, sft, immutable)
+      case HistogramByte       => readHistogram(input, sft, immutable)
+      case FrequencyByte       => readFrequency(input, sft, immutable, 2)
+      case Z3HistogramByte     => readZ3Histogram(input, sft, immutable, 2)
+      case Z3FrequencyByte     => readZ3Frequency(input, sft, immutable, 2)
+      case IteratorStackByte   => readIteratorStackCount(input, immutable)
+      case SeqStatByte         => readSeqStat(input, sft, immutable)
+      case FrequencyByteV1     => readFrequency(input, sft, immutable, 1)
+      case Z3HistogramByteV1   => readZ3Histogram(input, sft, immutable, 1)
+      case Z3FrequencyByteV1   => readZ3Frequency(input, sft, immutable, 1)
+      case DescriptiveStatByte => readDescriptiveStat(input, sft, immutable)
     }
   }
 
+  private [stats] def writeDescriptiveStats(output: Output, sft: SimpleFeatureType, stat: DescriptiveStats): Unit = {
+    val size = stat.size
+    output.writeInt(size, true)
+    stat.attributes.foreach(output.writeInt(_, true))
+    for(v <- stat._min.getMatrix.data) output.writeDouble(v)
+    for(v <- stat._max.getMatrix.data) output.writeDouble(v)
+    for(v <- stat._sum.getMatrix.data) output.writeDouble(v)
+    for(v <- stat._mean.getMatrix.data) output.writeDouble(v)
+    for(v <- stat._m2n.getMatrix.data) output.writeDouble(v)
+    for(v <- stat._m3n.getMatrix.data) output.writeDouble(v)
+    for(v <- stat._m4n.getMatrix.data) output.writeDouble(v)
+    for(v <- stat._c2.getMatrix.data) output.writeDouble(v)
+    output.writeLong(stat._count, true)
+  }
+
+  private [stats] def readDescriptiveStat(input: Input, sft: SimpleFeatureType, immutable: Boolean): DescriptiveStats = {
+    val size = input.readInt(true)
+    val attributes = for(_ <- 0 until size) yield input.readInt(true)
+
+    val stats = if (immutable) {
+      new DescriptiveStats(attributes) with ImmutableStat
+    } else {
+      new DescriptiveStats(attributes)
+    }
+
+    stats._min.getMatrix.setData((for(_ <- 0 until size) yield input.readDouble).toArray)
```

Comment:
oh, if size is just the number of attributes then it probably doesn't matter.
if we ever end up writing these stats to our metadata table we might want to revisit since they get serialized a lot then